hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/QueryUtil.java

```diff
@@ -267,30 +259,20 @@ private void initContext(InternalSerializationService iss) {
                 return;
             }
 
-            SecurityContext securityContext = node.securityContext;
-            if (securityContext != null) {
-                assert subject != null : "Missing subject when security context exists";
-                this.ssc = securityContext.createSqlContext(subject);
-            } else {
-                this.ssc = NoOpSqlSecurityContext.INSTANCE;
-            }
-
-            this.evalContext = ExpressionEvalContext.createContext(arguments, node.getNodeEngine(), iss, ssc);
+            this.evalContext = new UntrustedExpressionEvalContext(arguments, iss, node.nodeEngine);
```

Comment:
I can see that we do not have to pass UEEC when creating JoinProjection because MapProxyImpl uses serialization to clone the projection in `MapProxyImpl.project`: `projection = serializationService.toObject(serializationService.toData(projection))`. But IMO we still should create UEEC in similar way to `UpdatingEntryProcessor` so we do not depend on this behavior, which is undocumented and have consistent behavior when executing locally.

Reply:
I hope I understood you correctly. Fixed - Now you can create a JoinProjection only with UntrustedEvalExpression.

Reply:
✔️